mypy.ini

```diff
@@ -0,0 +1,25 @@
+[mypy]
+python_version = 3.8
+
+# Set some rather strict defaults, but also globally deactivate all errors
+# and then activate the typecheck only for certain modules.
+# The set of typechcecked modules can be incrementally expanded until everything
+# is correctly typed.
+warn_return_any = True
+warn_unused_configs = True
+disallow_untyped_defs = True
+disallow_any_generics = True
+disallow_any_decorated = True
+allow_redefinition = True
+warn_unreachable = True
+ignore_errors = True
```

Comment:
Include the description of each option, e.g.


```ini
# Allows variables to be redefined with an arbitrary type, as long as the
# redefinition is in the same block and nesting level as the original definition.
allow_redefinition = True
```

Reply:
I've added the descriptions. I've also removed `disallow_any_decorated`. The problem with that setting is, that it doesn't just warn if you have a typed definition that loses its type hints because of the decorator, it also warns if your original definition contains Any and has a decorator.

That means this pattern that appears in the codebase a lot generates warnings.
```
@staticmethod
def foo(a: str, b: int, **kwargs: Any): # kwargs is forwarded to some other function
    ...
```